drivers/smbus/Kconfig

```diff
@@ -38,6 +38,8 @@ config SMBUS_INTEL_PCH
 	bool "SMBus Intel PCH driver"
 	default y
 	depends on DT_HAS_INTEL_PCH_SMBUS_ENABLED
+	select PCIE
```

Comment:
Should PCIE maybe be enabled in the same way as drivers, ie based on a DTS node with a given compat being enabled? Or maybe we could introduce some DT_HAS_BUS_PCIE_ENABLED macro, since I suspect not all platforms have the same compat string for PCIe bus modes. Any thoughts on this from @galak or @mbolivar-nordic ?

Reply:
For this specific driver, Intel PCH, this is, probably, not needed. Only PCI is supported.
I can also move `select` `PCIE` and `DYNAMIC_INTERRUPTS` to `qemu_x86_64` board configuration.

Reply:
> Or maybe we could introduce some DT_HAS_BUS_PCIE_ENABLED macro

Would be easy to do if that makes sense